WooCommerce/Classes/Yosemite/DeauthenticatedState.swift

```diff
@@ -12,12 +11,20 @@ class DeauthenticatedState: StoresManagerState {
 
     /// Retains all of the active Services
     ///
-    private let services: [ActionsProcessor]
+    private let services: [DeauthenticatedStore]
 
     init() {
         services = [JetpackConnectionStore(dispatcher: dispatcher)]
     }
 
+    /// Asks the persisted stores to update their remote with the given siteURL and network.
+    ///
+    func updateStores(with siteURL: String, network: Network) {
+        for store in services {
+            store.updateRemote(with: siteURL, network: network)
```

Comment:
would any of the `DeauthenticatedStore`s work without the siteURL and network? if not, how about making `services: [DeauthenticatedStore]` a variable and then `remote: JetpackConnectionRemote` become a non-nil constant in `JetpackConnectionStore` and `DeauthenticatedStore.updateRemote` isn't necessary anymore?

Reply:
Even though siteURL and network are required for `JetpackConnectionStore`, I can't come up with a good solution to keep them in the store's initializer. We don't have any site URL and network immediately when initializing `DeauthenticatedState`,  only immediately when there's a need to work with WP.org API. That's why I decided to add this method to update the store later.

Please share your thought if you have a better solution for this.

Reply:
do we expect more than just `JetpackConnectionStore` that requires the same `siteURL` and `network` dependencies at a later time? if not, how about creating an action like `JetpackConnectionAction.update(siteURL: String, network: Network)` that initializes the `JetpackConnectionRemote` internally so that the async update is just an implementation detail within `JetpackConnectionStore`?

Reply:
Good idea! We may need to use WP.org API to install/activate Jetpack the plugin later but I suppose we can add those into this same `JetpackConnectionStore`. My only concern is that we'll need to initialize `WordPressOrgNetwork` from the UI layer when calling this action, but we can just import the class for that so I guess it's fine. Updated this in the later commits.

Reply:
> My only concern is that we'll need to initialize WordPressOrgNetwork from the UI layer when calling this action, but we can just import the class for that so I guess it's fine. Updated this in the later commits.

I'm not sure how the update action is used later, does the `Network` have to be initialized in the app/UI layer? Does it change over time together with `siteURL`? We can always see how we can improve the setup later when the actions are integrated with UI.